credentials/tls/certprovider/store.go

```diff
@@ -75,6 +93,28 @@ func (wp *wrappedProvider) Close() {
 	}
 }
 
+// Close overrides the Close method of the embedded provider to avoid release the
+// already released reference.
+func (w *singleCloseWrappedProvider) Close() {
+	newProvider := Provider(closedProvider{})
```

Comment:
Is this type cast required here?

Reply:
Yes. The type of the variable is important here (the next line uses a pointer to it).

But I can replace this line with:
```go
var newProvider Provider = closedProvider{}
```
or
```
var newProvider Provider
newProvider = closedProvider{}
```

But this is a taste. I do not know which is more consistent with your code style.